streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractStateManager.java

```diff
@@ -69,16 +69,15 @@ public void reinitializeStateStoresForPartitions(final Logger log,
             throw new StreamsException("Failed to reinitialize global store.", fatalException);
         }
 
-        final Iterator<Map.Entry<String, StateStore>> it = stateStores.entrySet().iterator();
-        while (it.hasNext()) {
-            final StateStore stateStore = it.next().getValue();
+        for (final Map.Entry<String, StateStore> entry : storesCopy.entrySet()) {
+            final StateStore stateStore = entry.getValue();
             final String storeName = stateStore.name();
             if (storeToBeReinitialized.contains(storeName)) {
                 try {
                     stateStore.close();
                 } catch (final RuntimeException ignoreAndSwallow) { /* ignore */ }
                 processorContext.uninitialize();
-                it.remove();
```

Comment:
I'm not sure why this would cause `ConcurrentModificationException`, isn't removing from the iterator (not from the map directly) allowed?

Reply:
The issue is triggered below in L101: `stateStore.init(processorContext, stateStore);` does add to `stateStores` again...

Reply:
Ah I see, thanks!